src/main/java/bio/terra/service/snapshot/Snapshot.java

```diff
@@ -180,4 +181,13 @@ public AzureRegion getStorageAccountRegion() {
   public boolean isSecureMonitoringEnabled() {
     return getFirstSnapshotSource().getDataset().isSecureMonitoringEnabled();
   }
+
+  public Map<String, Object> getMode() {
+    return mode;
```

Comment:
We don't always do this but something to consider is returning an immutable version of the field. A user can still modify the field via the setter but can't accidentally change it by modifying the result of the getter.
```suggestion
    return Map.copyOf(mode);
```